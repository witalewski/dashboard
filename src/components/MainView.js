import React, { Component } from 'react';
import styled from '@emotion/styled';
import { inject, observer } from 'mobx-react';
import { Section } from './Section';
import { ChartSection } from './sections/ChartSection';
import { SummarySection } from './sections/SummarySection';

const MainViewStyled = styled.main`
  display: flex;
  flex-direction: column;

  .container {
    display: flex;
    flex-wrap: wrap;
  }

  .summary-section {
    flex: 1 3 200px;
  }

  .monthly-sales-section {
    flex: 3 1 600px;
  }

  .most-popular-offers-section,
  .payment-methods-section,
  .devices-section {
    flex: 1 1 200px;
  }
`;

export class MainView extends Component {
  constructor(props) {
    super(props);
    this.state = { initialized: false };
  }

  componentDidMount() {
    if (!this.state.initialized) {
      this.props.loadMockChartData();
      this.setState({ initialized: true });
    }
  }

  render() {
    const { chartData } = this.props;
    return chartData ? (
      <MainViewStyled>
        <div className="container">
          <SummarySection
            className="summary-section"
            title="Summary"
            data={chartData.summaryChart.data}
          />
          <ChartSection
            title="Monthly Sales"
            className="monthly-sales-section"
            source={chartData.monthlySales}
            dataTransform={e => e}
          />
        </div>
        <div className="container">
          <ChartSection
            title="Most Popular Offers"
            className="most-popular-offers-section"
            source={chartData.mostPopular}
            dataTransform={e => e}
          />
          <ChartSection
            title="Payment Methods"
            className="payment-methods-section"
            source={chartData.paymentMethods}
            dataTransform={e => e}
          />
          <ChartSection
            title="Devices"
            className="devices-section"
            source={chartData.devices}
            dataTransform={e => e}
          />
        </div>
        <ChartSection
          title="Revenue This Month vs Revenue Previous Month"
          className="revenue-section"
          source={chartData.revenueMonthToMonth}
          dataTransform={({ x, y, y2 }) => ({
            x,
            'This month': y2,
            'Last month': y,
          })}
        />
        <Section title="Geography" className="geography-section">
          [Section Content]
        </Section>
      </MainViewStyled>
    ) : null;
  }
}

export default inject(({ appState }) => ({
  chartData: appState.chartData,
  loadMockChartData: appState.loadMockChartData,
}))(observer(MainView));
