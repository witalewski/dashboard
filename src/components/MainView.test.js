import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { MainView } from './MainView';

describe('MainView', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MainView />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
