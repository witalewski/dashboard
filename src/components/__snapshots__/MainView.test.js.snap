// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView matches snapshot out data 1`] = `null`;

exports[`MainView matches snapshot with mock data 1`] = `
<ForwardRef(render)>
  <div
    className="container"
  >
    <SummarySection
      className="summary-section"
      data={
        Object {
          "revenueAllTime": 262.22,
          "revenueAverage": 131.11,
          "revenueLastMonth": 199.09,
          "revenueRate": 0.38,
          "revenueThisMonth": 100.64,
        }
      }
      title="Summary"
    />
    <ChartSection
      className="monthly-sales-section"
      dataTransform={[Function]}
      source={
        Object {
          "bar1-label": "Event",
          "bar2-label": "Single/Rental",
          "bar3-label": "Pass",
          "bar4-label": "Subscription",
          "data": Array [
            Object {
              "bar1": 120.65,
              "bar2": null,
              "bar3": 6562.67,
              "bar4": null,
              "label": "December",
              "line1": 12455,
            },
            Object {
              "bar1": 243.64,
              "bar2": null,
              "bar3": 7165.2,
              "bar4": null,
              "label": "January",
              "line1": 13468,
            },
            Object {
              "bar1": 345.54,
              "bar2": 12.4,
              "bar3": 46744.11,
              "bar4": null,
              "label": "February",
              "line1": 58446,
            },
          ],
          "dataType": "bar",
          "line1-label": "Transactions",
        }
      }
      title="Monthly Sales"
    />
  </div>
  <div
    className="container"
  >
    <ChartSection
      className="most-popular-offers-section"
      dataTransform={[Function]}
      source={
        Object {
          "data": Array [
            Object {
              "label": "E946347906_AU",
              "tooltip": "Brian Foyle Street Stock Memorial
Revenue: $33",
              "value": 990,
            },
            Object {
              "label": "E643516498_AU",
              "tooltip": "World Series Sprintcars -USC - NSW Rnd 8
Revenue: $1,564.30",
              "value": 105,
            },
            Object {
              "label": "E657734674_AU",
              "tooltip": "USC-NSW Rnd 7
Revenue: $15,347.46",
              "value": 456,
            },
            Object {
              "label": "E135478956_AU",
              "tooltip": "Sprintcar Allstars @ MMS
Revenue: $35,441.54",
              "value": 1204,
            },
          ],
          "dataType": "pie",
        }
      }
      title="Most Popular Offers"
    />
    <ChartSection
      className="payment-methods-section"
      dataTransform={[Function]}
      source={
        Object {
          "data": Array [
            Object {
              "label": "VISA",
              "tooltip": "Revenue: $339,938.46",
              "value": 12125,
            },
            Object {
              "label": "MasterCard",
              "tooltip": "Revenue: $220,377.81",
              "value": 7880,
            },
            Object {
              "label": "PayPal",
              "tooltip": "Revenue: $221,042.77",
              "value": 8092,
            },
            Object {
              "label": "iDeal",
              "tooltip": "Revenue: $120.6",
              "value": 15,
            },
          ],
          "dataType": "pie",
        }
      }
      title="Payment Methods"
    />
    <ChartSection
      className="devices-section"
      dataTransform={[Function]}
      source={
        Object {
          "data": Array [
            Object {
              "label": "Desktop",
              "tooltip": null,
              "value": 6013,
            },
            Object {
              "label": "Tablet",
              "tooltip": null,
              "value": 1577,
            },
            Object {
              "label": "Smartphone",
              "tooltip": null,
              "value": 1252,
            },
            Object {
              "label": "TV",
              "tooltip": null,
              "value": 404,
            },
          ],
          "dataType": "pie",
        }
      }
      title="Devices"
    />
  </div>
  <ChartSection
    className="revenue-section"
    dataTransform={[Function]}
    source={
      Object {
        "data": Array [
          Object {
            "x": 1,
            "y": 2156.75,
            "y2": 34997.48,
          },
          Object {
            "x": 2,
            "y": 4688.6,
            "y2": 34997.12,
          },
          Object {
            "x": 3,
            "y": 12045.43,
            "y2": 49449.76,
          },
          Object {
            "x": 4,
            "y": 33516.68,
            "y2": 62774.3,
          },
        ],
        "dataType": "line2",
        "x-label": "Days of Month",
        "y-label": "Previous Month",
        "y2-label": "This Month",
      }
    }
    title="Revenue This Month vs Revenue Previous Month"
  />
  <Section
    className="geography-section"
    title="Geography"
  >
    [Section Content]
  </Section>
</ForwardRef(render)>
`;
